### DISCUSSION TITLE
No external editors found. Install vs code?

### DISCUSSION THREAD

**Discussion by Rubelefsky** [community/community#24073] [Apps, API and Webhooks]
<p>Just downloaded GitHub desktop for my PC and MAC. I Have sublime text 4 on each.  On my Mac it recognizes sublime, pycharm etc.</p>
<p>On my PC however it does not recognize any external editor and asks if I want to install VS Code.</p>
<p>I am new to Github and Github desktop.</p>

**Comment by steveward** [comment, marked as answer]
<p>Welcome <a class="mention" href="/vegasaan">@Vegasaan</a>!</p>
<p>At this time GitHub Desktop only supports editors that have been specifically integrated into the application. You can find the details of how this works along with the supported editors here:</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/desktop/desktop/blob/be4534ca7079c671b4d1381af089a4ccbba98d27/docs/technical/editor-integration.md" target="_blank" rel="noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/desktop/desktop/blob/be4534ca7079c671b4d1381af089a4ccbba98d27/docs/technical/editor-integration.md" target="_blank" rel="noopener">desktop/desktop/blob/be4534ca7079c671b4d1381af089a4ccbba98d27/docs/technical/editor-integration.md</a></h4>
<pre><code class="lang-md"># "Open External Editor" integration

GitHub Desktop supports the user choosing an external program to open their
local repositories, and this is available from the top-level **Repository** menu
or when right-clicking on a repository in the sidebar.

### My favourite editor XYZ isn't supported!

This is the checklist of things that it needs to support:

 - the editor supports opening a directory, not just a file
 - the editor is installed by the user, so there is a reliable way to find it
   on the user's machine
 - it comes with a command-line interface that can be launched by Desktop

If you think your editor satisfies all these please read on to understand how
Desktop integrates with each OS, and if you're still keen to integrate this
please fork and contribute a pull request for the team to review.

## Windows
</code></pre>

  This file has been truncated. <a href="https://github.com/desktop/desktop/blob/be4534ca7079c671b4d1381af089a4ccbba98d27/docs/technical/editor-integration.md" target="_blank" rel="noopener">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>At this time PyCharm is not supported on Windows, but Sublime Text should work. Since Sublime Text 4 is new it is possible that something has changed to break this. I’ll take a look to see if I can reproduce this issue.</p>

**Comment by Rubelefsky** [comment]
<p>Yeah it may just be windows because on my Mac it supports PyCharm and ST4</p>

**Comment by ooksanen** [comment]
<p>Can confirm, Sublime Text 4 doesn’t show up in available editors on Windows, even after uninstalling ST3. Would be brilliant to have ST4 as an option too.</p>
<p>EDIT: I found a dirtyish workaround. You can update from ST3 (<a href="https://www.sublimetext.com/3" rel="noopener nofollow ugc">download from here</a>) to ST4 which should keep the registry keys and such  in the ST3 format GitHub Desktop expects and you can open repos in ST4.</p>

**Comment by stephenteacher** [comment]
<p>I have notepadd++ as a portable app (portable <a href="http://apps.com" rel="noopener nofollow ugc">apps.com</a>) and, I may be wrong here, I don’t <em>think</em> that portable apps create any registry keys,  I can’t check as I’m not Admin… (I have also move Atom to my portable apps directory).  Is there a way I could point the Github desktop program to that directory?</p>

**Comment by steveward** [comment]
<p><a class="mention" href="/stephenteacher">@stephenteacher</a> unfortunately we don’t have a way of pointing GitHub Desktop to portable editor installs at this time.</p>
